packages/legend-graph/src/models/metamodels/pure/test/Testable.ts

```diff
@@ -16,31 +16,49 @@
 
 import { isNonNullable } from '@finos/legend-shared';
 import type { PureModel } from '../../../../graph/PureModel';
+import type { GraphManagerState } from '../../../../GraphManagerState';
 import { PackageableElement } from '../packageableElements/PackageableElement';
 import type { Test } from './Test';
 
 export interface Testable {
   tests: Test[];
 }
-export const getNullableTestable = (
+
+export type TestableIDBuilder = (
+  testable: Testable,
+  graph: PureModel,
+) => string | undefined;
+
+export type TestableFinder = (
   id: string,
   graph: PureModel,
+) => Testable | undefined;
+
+export type TestablesGetter = (graph: PureModel) => Testable[];
+
+export const getNullableTestable = (
+  id: string,
+  graphManager: GraphManagerState,
```

Comment:
we can't pass a state here, so instead, let's pass the `graph` and the list of `PureGraphManager` plugins.

Also, we should not have logic in `metamodel` so let's move this out to `src/helper/Testable_Helper.ts`